src/main/java/com/mojang/brigadier/StringReader.java

```diff
@@ -239,4 +239,30 @@ public void expect(final char c) throws CommandSyntaxException {
         }
         skip();
     }
+
+    public void expect(final String s) throws CommandSyntaxException {
+        if (!canRead(s.length()) || !getRemaining().startsWith(s)) {
+            throw CommandSyntaxException.BUILT_IN_EXCEPTIONS.readerExpectedSymbol().createWithContext(this, s);
+        }
+        this.cursor += s.length();
+    }
+
+    public String expectOption(final String... opts) throws CommandSyntaxException {
+        String out = "";
+        final int start = this.getCursor();
+        for (String s: opts) {
+            try {
+                this.setCursor(start);
+                expect(s);
+                if (out.length() < s.length()) {
+                    out = s;
+                }
+            } catch (final CommandSyntaxException ignored) {}
+        }
+        if (out.length() == 0) {
```

Comment:
If one of the `opts` is `""`, this will inadvertantly throw an error. Please add a unit test to cover this - and either throw a deliberate "you can't expect nothing" error (IllegalArgumentException?) or handle this properly (default to null, test for null for your match)